package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -388,79 +316,147 @@
    :inherited-members:
 
 """
-from MDAnalysis.lib.correlations import autocorrelation, correct_intermittency
-import MDAnalysis.analysis.hbonds
 from itertools import zip_longest
 import logging
-import warnings
+
 import numpy as np
 
+from MDAnalysis.lib.log import ProgressBar
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.lib.correlations import autocorrelation, correct_intermittency
 
 logger = logging.getLogger('MDAnalysis.analysis.waterdynamics')
-from MDAnalysis.lib.log import ProgressBar
 
 
-class WaterOrientationalRelaxation(object):
+class WaterOrientationalRelaxation(AnalysisBase):
     r"""Water orientation relaxation analysis
 
-    Function to evaluate the Water Orientational Relaxation proposed by Yu-ling
-    Yeh and Chung-Yuan Mou [Yeh1999_]. WaterOrientationalRelaxation indicates
-    "how fast" water molecules are rotating or changing direction. This is a
-    time correlation function given by:
+    Class evaluating the water orientational relaxation proposed by Yu-ling
+    Yeh and Chung-Yuan Mou [Yeh1999_].
+    `WaterOrientationalRelaxation` (WOR) indicates
+    "how fast" water molecules are rotating or changing direction.
+    If WOR is very stable we can assume that water molecules are
+    rotating/changing direction very slowly, on the
+    other hand, if WOR decay very fast, we can assume that water molecules are
+    rotating/changing direction very fast. The results are
+    time correlation functions given by:
 
     .. math::
         C_{\hat u}(\tau)=\langle \mathit{P}_2[\mathbf{\hat{u}}(t_0)\cdot\mathbf{\hat{u}}(t_0+\tau)]\rangle
 
-    where :math:`P_2=(3x^2-1)/2` is the second-order Legendre polynomial and :math:`\hat{u}` is
-    a unit vector along HH, OH or dipole vector.
-
+    where :math:`P_2=(3x^2-1)/2` is the second-order Legendre polynomial
+    and :math:`\hat{u}` is a unit vector along HH, OH or dipole vector.
 
     Parameters
     ----------
     universe : Universe
       Universe object
     selection : str
       Selection string for water [‘byres name OH2’].
-    t0 : int
-      frame  where analysis begins
-    tf : int
-      frame where analysis ends
-    dtmax : int
-      Maximum dt size, `dtmax` < `tf` or it will crash.
```

Comment:
@alejob what is `dtmax` supposed to be doing?

Is it the same as a stride when you subselect frames from a trajectory as in `trajectory[start:stop:step]`? If not,  please explain.

Reply:
It is not the same as trajectory[start:stop:step].

First, we need to define `dt`. `dt` is the delta time in which WOR is calculated. For instance, if `dt=5` this means that WOR will be calculated comparing frame 1 with frame 5, and then comparing frame 6 with 10, then 11 with 15, and so on. Then we get the mean of these values and get the final value for `dt=5`. If `dt=8`, WOR will be calculated between frame 1 and frame 8, between 9 and 16 and repeat all the process. Now, `dtmax` indicates the maximum value that `dt` could have, and indicates that all the minor `dt` will be calculated, in other words, if you plot the output of this analysis, `dtmax` is the maximum value of the `x` axis. 

Here an example modified from the docs:
If you have that `t0 = 0`, `tf = 1000` and `dtmax = 20`,  we create 20 windows
timesteps (20 values in the x axis), the first window is created with 1000
timestep average (1000/1), the second window is created with 500 timestep
average(1000/2), the third window is created with 333 timestep average (1000/3)
and so on. Obtaining the mean of many timesteps gives us a better statistic for the WOR value, _i.e_, a smoother curve.

Because of this fix `dtmax` in `dtmax=2` has non sense, `dtmax` should be chosen by the user.